crate/operator/main.py

```diff
@@ -592,3 +593,41 @@ async def update_cratedb_resource(
                     await ensure_user_password_label(
                         core, namespace, user_spec["password"]["secretKeyRef"]["name"]
                     )
+
+
+@kopf.on.update("", "v1", "secrets", labels={LABEL_USER_PASSWORD: "true"})
+async def secret_update(
+    namespace: str,
+    name: str,
+    diff: kopf.Diff,
+    **kwargs,
+):
+    for operation, field_path, old_value, new_value in diff:
+        if field_path == ("data", "password"):
+            async with ApiClient() as api_client:
+                coapi = CustomObjectsApi(api_client)
+                core = CoreV1Api(api_client)
+
+                custom_objects = await coapi.list_namespaced_custom_object(
+                    namespace=namespace,
+                    group=API_GROUP,
+                    version="v1",
+                    plural=RESOURCE_CRATEDB,
+                )
+
+                for crate_custom_object in custom_objects["items"]:
+                    crate_custom_object_users = crate_custom_object["spec"]["users"]
+
+                    updated_user_spec = None
+                    for user_spec in crate_custom_object_users:
+                        if name == user_spec["password"]["secretKeyRef"]["name"]:
+                            updated_user_spec = user_spec
+
+                    if updated_user_spec is not None:
+                        await update_user_password(
```

Comment:
Note to myself: Check if the same secret can be used in the same CrateDB resource for multiple users.

Reply:
I think a secret **can** be used in multiple users of one CrateDB resource. Therefore, I updated the logic.

Reply:
Yes, it can.